pkg/event/consumer.go

```diff
@@ -0,0 +1,238 @@
+package event
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/content-services/content-sources-backend/pkg/event/message"
+	"github.com/content-services/content-sources-backend/pkg/event/schema"
+	"github.com/rs/zerolog/log"
+)
+
+// https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+// https://docs.confluent.io/platform/current/clients/consumer.html#ak-consumer-configuration
+// TODO Load Consumer Configmap from a file indicated by
+//      KAFKA_CONSUMER_CONFIG_FILE (for clowder it will be a secret, for local
+//      workstation will be the ${PROJECT_DIR}}/configs/kafka-consumer.yaml)
+
+// NewConsumer create a new consumer based on the configuration
+// supplied.
+// config Provide the necessary configuration to create the consumer.
+
+func NewConsumer(config *KafkaConfig) (*kafka.Consumer, error) {
+	var (
+		consumer *kafka.Consumer
+		err      error
+	)
+
+	if config == nil {
+		return nil, fmt.Errorf("config cannot be nil")
+	}
+
+	// TODO Refactor the KafkaConfig structure will impact here
+	kafkaConfigMap := &kafka.ConfigMap{
+		"bootstrap.servers":        config.Bootstrap.Servers,
+		"group.id":                 config.Group.Id,
+		"auto.offset.reset":        config.Auto.Offset.Reset,
+		"auto.commit.interval.ms":  config.Auto.Commit.Interval.Ms,
+		"go.logs.channel.enable":   false,
+		"allow.auto.create.topics": true,
+		// NOTE This could be useful when launching locally
+		// "socket.timeout.ms":                  60000,
+		// "socket.connection.setup.timeout.ms": 3000,
+		// "session.timeout.ms":                 6000,
+	}
+
+	if config.Sasl.Username != "" {
+		_ = kafkaConfigMap.SetKey("sasl.username", config.Sasl.Username)
+		_ = kafkaConfigMap.SetKey("sasl.password", config.Sasl.Password)
+		_ = kafkaConfigMap.SetKey("sasl.mechanism", config.Sasl.Mechanism)
+		_ = kafkaConfigMap.SetKey("security.protocol", config.Sasl.Protocol)
+		_ = kafkaConfigMap.SetKey("ssl.ca.location", config.Capath)
+	}
+
+	if consumer, err = kafka.NewConsumer(kafkaConfigMap); err != nil {
+		return nil, err
+	}
+
+	if err = consumer.SubscribeTopics(config.Topics, nil); err != nil {
+		return nil, err
+	}
+	log.Info().Msgf("Consumer subscribed to topics: %s", strings.Join(config.Topics, ","))
+
+	return consumer, nil
+}
+
+func getHeader(msg *kafka.Message, key string) (*kafka.Header, error) {
+	if msg == nil {
+		return nil, fmt.Errorf("msg is nil")
+	}
+	if key == "" {
+		return nil, fmt.Errorf("key is empty")
+	}
+	for _, header := range msg.Headers {
+		if header.Key == key {
+			return &header, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find '%s' in message header", key)
+}
+
+// TODO Refactor to make this check dynamic so it does not need
+//      to be modified if more different messages are added
+func isValidEvent(event string) bool {
```

Comment:
eh, i think we can worry about that refactor when we have a new event type we're listening for